### DISCUSSION TITLE
Translations

### DISCUSSION THREAD

**Discussion by milotype** [vufind-org/vufind#2500] [General]
If possible, please consider using a translation platform ([Weblate](https://weblate.org/hr/), [Transifex](https://www.transifex.com/), [POEditor](https://poeditor.com/), ...) for translations. My preference is Weblate, but of course it all depends what suits the **vufind** project best.

Currently it is very tedious to find the updated/new string changes in the `.ini` files.

I'm just a translator, so I don't know if it's possible to implement it.

**Comment by demiankatz** [comment]
@milotype, VuFind includes a language report which can quickly identify all missing strings that need to be translated. I use this as part of the release process when I reach out to translators to request updates. I have added some quick documentation about it in [the wiki localization page](development:architecture:localization#updating_a_language).

If you don't have a running instance of VuFind, you can always email me to request a report -- I don't want to waste your time doing manual work if it is not necessary!

We have considered a translation platform in the past, but previous attempts failed due to lack of communication from translation platform maintainers. It's certainly a topic worth revising, especially if there is a service geared toward open source projects that would not charge us (since, at the moment, the project has no budget to pay for such services). I'll put this on the agenda of a future Project Management Committee meeting for further discussion.

**Reply by milotype** [reply]
@demiankatz I don't run VuFind, I'm just a translator who from time to time checkes translations of projects I supported in the past. 
Until you decide upon a translation platform, maybe you can find a way to inform translators per email ...

**Reply by demiankatz** [reply]
@milotype, it's already part of my release workflow to inform all of the volunteer translators (including you) about new strings -- this usually happens about once every six months. But if you would prefer to do smaller chunks of work more frequently, I'd certainly be happy to work with you on a schedule that is more convenient for your needs.

**Reply by milotype** [reply]
@demiankatz You can inform me however you feel appropriate.

**Reply by demiankatz** [reply]
Great, I appreciate your ongoing support!

**Comment by milotype** [comment]
@demiankatz Can you please talk about using a translation platform at your **Project Management Committee meeting**.
Three years have pased since my first proposal to use a translation platform (as mentioned, Weblate is my favorite).

I don't think that your project is too large for a free plan, but you should inspect the options!

Translation platforms have the benefit to:
- automatically update translations
- use a glossary for consistent translation
- see which strings need correction/translation

**Reply by demiankatz** [reply]
We use Lokalise now as a direct result of your suggestion. I thought you were already aware; I'm sorry if there was a failure of communication. If you need an invitation to the platform, please let me know.

**Comment by milotype** [comment]
@demiankatz You are right! Sorry, I completely forgot that I have an account on Lokalise.
